checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -699,8 +749,8 @@ private void addAnnotationForLiteralPlus(
                 return;
             }
             if (val < -1) {
-                if (nonLiteralType.hasAnnotation(LTLengthOf.class)
```

Comment:
To keep the diff focused, I suggest undoing this change and saving it for a separate pull request if it is needed.